### PR TITLE
Ruby nested msg caps

### PR DESCRIPTION
**Pull request by toc007**
#4852
Capitalize all prefix while printing out all nested messages.

### CODE REVIEW

**Comment by toc007** [comment]
I signed it!

**Comment by TeBoring** [comment]
Would you mind adding a test?

File: ruby/tests/generated_code.proto [hunk 1]
```diff
@@ -53,6 +53,13 @@ message TestMessage {
   map<string, bool> map_string_bool = 70;
 
   message NestedMessage {
+    message Hello {
```
Comment (TeBoring) [inline ruby/tests/generated_code.proto]: Does this cause problem without your fix? I thought a lower letter started name causes the problem.

File: ruby/tests/generated_code.proto [hunk 2]
```diff
@@ -69,6 +69,12 @@ enum TestEnum {
   C = 3;
 }
 
+message TestNestedMsg {
```
Comment (TeBoring) [inline ruby/tests/generated_code.proto]: Could you give it a better name? e.g.,
```
message testlowercase {
  message lowercase {}
}
message _test_underscore {
  message _underscore {}
}
```

File: ruby/tests/generated_code.proto [hunk 3]
```diff
@@ -69,6 +69,13 @@ enum TestEnum {
...
```
[the remark this answers is not in the document]
Comment (toc007) [reply ruby/tests/generated_code.proto]: Would you like me to have testLowercase as the top-level message, or get rid of TestNestedMsg entirely?

**Pull request merged**